Index: src/main/java/ru/javaops/masterjava/xml/util/StaxStreamProcessor.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javaops/masterjava/xml/util/StaxStreamProcessor.java	(date 1489431163000)
+++ src/main/java/ru/javaops/masterjava/xml/util/StaxStreamProcessor.java	(revision )
@@ -24,15 +24,26 @@
     }
 
     public boolean doUntil(int stopEvent, String value) throws XMLStreamException {
+        return doUntilAny(stopEvent, value) != null;
+    }
+
+    public String getAttribute(String name) throws XMLStreamException {
+        return reader.getAttributeValue(null, name);
+    }
+
+    public String doUntilAny(int stopEvent, String... values) throws XMLStreamException {
         while (reader.hasNext()) {
             int event = reader.next();
             if (event == stopEvent) {
-                if (value.equals(getValue(event))) {
-                    return true;
+                String xmlValue = getValue(event);
+                for (String value : values) {
+                    if (value.equals(xmlValue)) {
+                        return xmlValue;
+                    }
                 }
             }
         }
-        return false;
+        return null;
     }
 
     public String getValue(int event) throws XMLStreamException {
Index: src/test/java/ru/javaops/masterjava/MainXml.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/java/ru/javaops/masterjava/MainXml.java	(date 1489431163000)
+++ src/test/java/ru/javaops/masterjava/MainXml.java	(revision )
@@ -1,5 +1,6 @@
 package ru.javaops.masterjava;
 
+import com.google.common.base.Splitter;
 import com.google.common.io.Resources;
 import j2html.tags.ContainerTag;
 import one.util.streamex.StreamEx;
@@ -9,7 +10,9 @@
 import ru.javaops.masterjava.xml.schema.User;
 import ru.javaops.masterjava.xml.util.JaxbParser;
 import ru.javaops.masterjava.xml.util.Schemas;
+import ru.javaops.masterjava.xml.util.StaxStreamProcessor;
 
+import javax.xml.stream.events.XMLEvent;
 import java.io.InputStream;
 import java.io.Writer;
 import java.net.URL;
@@ -18,6 +21,7 @@
 import java.util.*;
 import java.util.stream.Collectors;
 
+import static com.google.common.base.Strings.nullToEmpty;
 import static j2html.TagCreator.*;
 
 
@@ -43,6 +47,10 @@
         try (Writer writer = Files.newBufferedWriter(Paths.get("out/users.html"))) {
             writer.write(html);
         }
+
+        System.out.println();
+        users = processByStax(projectName, payloadUrl);
+        users.forEach(System.out::println);
     }
 
     private static Set<User> parseByJaxb(String projectName, URL payloadUrl) throws Exception {
@@ -64,6 +72,46 @@
                 .collect(Collectors.toCollection(() -> new TreeSet<>(USER_COMPARATOR)));
     }
 
+    private static Set<User> processByStax(String projectName, URL payloadUrl) throws Exception {
+
+        try (InputStream is = payloadUrl.openStream()) {
+            StaxStreamProcessor processor = new StaxStreamProcessor(is);
+            final Set<String> groupNames = new HashSet<>();
+
+            // Projects loop
+            projects:
+            while (processor.doUntil(XMLEvent.START_ELEMENT, "Project")) {
+                if (projectName.equals(processor.getAttribute("name"))) {
+                    // Groups loop
+                    String element;
+                    while ((element = processor.doUntilAny(XMLEvent.START_ELEMENT, "Project", "Group", "Users")) != null) {
+                        if (!element.equals("Group")) {
+                            break projects;
+                        }
+                        groupNames.add(processor.getAttribute("name"));
+                    }
+                }
+            }
+            if (groupNames.isEmpty()) {
+                throw new IllegalArgumentException("Invalid " + projectName + " or no groups");
+            }
+
+            // Users loop
+            Set<User> users = new TreeSet<>(USER_COMPARATOR);
+
+            while (processor.doUntil(XMLEvent.START_ELEMENT, "User")) {
+                String groupRefs = processor.getAttribute("groupRefs");
+                if (!Collections.disjoint(groupNames, Splitter.on(' ').splitToList(nullToEmpty(groupRefs)))) {
+                    User user = new User();
+                    user.setEmail(processor.getAttribute("email"));
+                    user.setValue(processor.getText());
+                    users.add(user);
+                }
+            }
+            return users;
+        }
+    }
+
     private static String toHtml(Set<User> users, String projectName) {
         final ContainerTag table = table().with(
                 tr().with(th("FullName"), th("email")))
