Index: services/mail-service/src/main/java/ru/javaops/masterjava/service/mail/MailServiceExecutor.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- services/mail-service/src/main/java/ru/javaops/masterjava/service/mail/MailServiceExecutor.java	(date 1494640683000)
+++ services/mail-service/src/main/java/ru/javaops/masterjava/service/mail/MailServiceExecutor.java	(revision )
@@ -1,6 +1,5 @@
 package ru.javaops.masterjava.service.mail;
 
-import com.google.common.collect.ImmutableList;
 import lombok.extern.slf4j.Slf4j;
 import one.util.streamex.StreamEx;
 import ru.javaops.masterjava.service.mail.util.MailUtils;
@@ -73,8 +72,7 @@
         addressees.forEach(addressee ->
                 mailExecutor.submit(() -> {
                     try {
-                        MailSender.sendTo(addressee, mailObject.getSubject(), mailObject.getBody(),
-                                ImmutableList.of(MailUtils.getAttach(mailObject.getAttachName(), mailObject.getAttachData())));
+                        MailSender.sendTo(addressee, mailObject.getSubject(), mailObject.getBody(), MailUtils.getAttaches(mailObject.getAttaches()));
                     } catch (WebStateException e) {
                         // already logged
                     }
Index: services/mail-api/src/main/java/ru/javaops/masterjava/service/mail/util/MailUtils.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- services/mail-api/src/main/java/ru/javaops/masterjava/service/mail/util/MailUtils.java	(date 1494640683000)
+++ services/mail-api/src/main/java/ru/javaops/masterjava/service/mail/util/MailUtils.java	(revision )
@@ -6,7 +6,6 @@
 import com.sun.istack.NotNull;
 import lombok.AllArgsConstructor;
 import lombok.Data;
-import org.apache.commons.io.IOUtils;
 import org.apache.commons.io.input.CloseShieldInputStream;
 import ru.javaops.masterjava.service.mail.Addressee;
 import ru.javaops.masterjava.service.mail.Attach;
@@ -14,7 +13,10 @@
 import javax.activation.DataHandler;
 import javax.activation.DataSource;
 import java.io.*;
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 public class MailUtils {
 
@@ -29,16 +31,12 @@
         private @NotNull String users;
         private String subject;
         private @NotNull String body;
-        private String attachName;
-        private byte[] attachData;
+        //  http://stackoverflow.com/questions/521171/a-java-collection-of-value-pairs-tuples
+        private List<SimpleImmutableEntry<String, byte[]>> attaches;
     }
 
-    public static MailObject getMailObject(String users, String subject, String body, String name, InputStream is) {
-        try {
-            return new MailObject(users, subject, body, name, is == null ? null : IOUtils.toByteArray(is));
-        } catch (IOException e) {
-            throw new IllegalStateException(e);
-        }
+    public static List<Attach> getAttaches(List<SimpleImmutableEntry<String, byte[]>> attaches) {
+        return attaches.stream().map(a -> getAttach(a.getKey(), a.getValue())).collect(Collectors.toList());
     }
 
     public static Attach getAttach(String name, byte[] attachData) {
Index: web/webapp/src/main/java/ru/javaops/masterjava/webapp/JmsSendServlet.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- web/webapp/src/main/java/ru/javaops/masterjava/webapp/JmsSendServlet.java	(date 1494640683000)
+++ web/webapp/src/main/java/ru/javaops/masterjava/webapp/JmsSendServlet.java	(revision )
@@ -1,7 +1,8 @@
 package ru.javaops.masterjava.webapp;
 
+import com.google.common.collect.ImmutableList;
 import lombok.extern.slf4j.Slf4j;
-import ru.javaops.masterjava.service.mail.util.MailUtils;
+import org.apache.commons.io.IOUtils;
 import ru.javaops.masterjava.service.mail.util.MailUtils.MailObject;
 
 import javax.jms.*;
@@ -16,6 +17,8 @@
 import javax.servlet.http.Part;
 import java.io.IOException;
 import java.lang.IllegalStateException;
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.List;
 
 @WebServlet("/sendJms")
 @Slf4j
@@ -55,13 +58,16 @@
         req.setCharacterEncoding("UTF-8");
         Part filePart = req.getPart("attach");
 
-        MailObject mailObject = MailUtils.getMailObject(
-                req.getParameter("users"),
-                req.getParameter("subject"),
-                req.getParameter("body"),
-                filePart == null ? null : filePart.getSubmittedFileName(),
-                filePart == null ? null : filePart.getInputStream());
+        List<SimpleImmutableEntry<String, byte[]>> attaches;
+        if (filePart.getSize() == 0) {
+            attaches = ImmutableList.of();
+        } else {
+            attaches = ImmutableList.of(
+                    new SimpleImmutableEntry<>(filePart.getSubmittedFileName(), IOUtils.toByteArray(filePart.getInputStream()))
+            );
+        }
 
+        MailObject mailObject = new MailObject(req.getParameter("users"), req.getParameter("subject"), req.getParameter("body"), attaches);
         resp.getWriter().write(sendJms(mailObject));
     }
 
